flow_action_components/SendHTMLEmail/force-app/main/default/classes/SendHTMLEmail.cls

```diff
@@ -4,38 +4,41 @@
  * @Author             : various, including Salesforce, Alex Edelstein, https://digitalflask.com/blog/send-email-attachments-salesforce-apex/
  * @Group              : unofficialSF
  * @Last Modified By   : jack.pond@psitex.com
- * @Last Modified On   : 2/5/2020, 4:02:33 PM
+ * @Last Modified On   : 3/22/2020, 9:08:33 AM
  * @Modification Log   : 
  * @License            : LICENSE found in https://github.com/alexed1/LightningFlowComponents
  * Ver       Date            Author      		    Modification
- * 1.32    2/5/2020   jack.pond@psitex.com     Modified for multi-lingual and to throw InvocableActionException on exceptions
+ * 1.32    2/5/2020   jack.pond@psitex.com      Modified for multi-lingual and to throw InvocableActionException on exceptions
+ * 1.32    2/11/2020  alex@edelstein.org        Bulkified and Changed recordId to String
+ * 1.33    3/22/2020  jack.pond@psitex.com      Added consistency checks, modified Labels and alphabetized params
 **/
 
 public without sharing class SendHTMLEmail {
 
     @invocableMethod(label='Send HTML Email')
-    public static List<Response>  SendEmail(List<Request> requests) {
+    public static List<Response> SendEmail(List<Request> requests) {
 
         List<Response> responseList = new List<Response>();
-
+		system.debug('Got here SendEmail');
         for (Request curRequest : requests) {
             Response response = new Response();
 
             String HTMLbody = curRequest.HTMLbody;
+            String orgWideEmailAddressId = curRequest.orgWideEmailAddressId;
             String plainTextBody = curRequest.plainTextBody;
-            String subject = curRequest.subject;
+            String recordId = curRequest.recordId;
             String replyEmailAddress = curRequest.replyEmailAddress;
             String senderDisplayName = curRequest.senderDisplayName;
+            String subject = curRequest.subject;
             String templateID = curRequest.templateID;
             String templateName = curRequest.templateName;
             String templateLanguage = curRequest.templateLanguage;
             String templateTargetObjectId = curRequest.templateTargetObjectId;
-            String orgWideEmailAddressId = curRequest.orgWideEmailAddressId;
-            Boolean saveAsActivity = curRequest.saveAsActivity;
-            String recordId = curRequest.recordId;
-
-
-
+            Boolean useSalesforceSignature = curRequest.useSalesforceSignature == null?true:curRequest.useSalesforceSignature;
```

Comment:
Is it possible for the SalesforceSignature to be undefined? If so, would htat get the user into trouble by default here?

Reply:
Great point - debated this one.  In one of the examples, when left null, I got an error - so i added this to use the default as specified in [setUseSignature(useSignature)](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_email_outbound_base.htm#apex_Messaging_Email_setUseSignature) (True).  This is probably a bug in the methods for either the [Email Class](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_email_outbound_base.htm) or the [SingleEmailMessage Methods](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_email_outbound_single.htm).  Regardless, I figured it would safe at least a couple of comments on the documentation stream asking for help in that condition.